api/test/helpers.ts

```diff
@@ -1,16 +1,12 @@
-import { getCurrentDate, parseDate } from "@shared/dateHelpers";
-import { defaultDateFormat } from "@shared/defaultConstants";
+import { TaxFiling } from "@shared/taxFiling";
 import { createHmac } from "node:crypto";
+import { calculateNextAnnualFilingDates } from "../src/domain/annual-filings/calculateNextAnnualFilingDates";
 
-export const determineAnnualFilingDate = (dateOfFormation: string) => {
-  const currentDate = getCurrentDate();
-  const dateOfFormationDate = parseDate(dateOfFormation);
-  let year = currentDate.year();
-  if (dateOfFormationDate.month() < currentDate.month()) {
-    year = year + 1;
-  }
-  const nextMonth = dateOfFormationDate.month() + 2;
-  return parseDate(`${year}-${nextMonth}-01`).add(-1, "day").format(defaultDateFormat);
+export const generateAnnualFilings = (dateOfFormation: string): TaxFiling[] => {
+  return calculateNextAnnualFilingDates(dateOfFormation).map((dueDate: string) => ({
+    identifier: "ANNUAL_FILING",
+    dueDate,
+  }));
```

Comment:
I would tend against having a test-helper that calls the real function, but it winds up being a sort of self-fulfilling prophecy of a test.

I wonder if a simpler implementation of a helper like this that we could use in all the places you need it, would be a function that takes an array of 3 dates and just transforms them into the proper taxFiling object. But doesn't do any calculation logic.

That way, it's the responsibility of the caller to define what the expected dates are, which is part of making the tests more specific and deterministic

Reply:
Yeah we talked about it for a while, the old helper function was just duplicating the logic of the the real function and given how this helper is being used, it seems appropriate to just use the real logic here. To do what you're talking about we'd have to mock the calculateNextAnnualFilingDates function itself, right?

Reply:
Yeah, I can see that the old implementation was also duplicating the logic. I don't think that's good, and this seems like an opportunity to improve it ("clean up the campground")

I'm not suggesting mocking the function, I'm imagining something simple like:
```
export const generateAnnualFilings = (dueDates: string[]): TaxFiling[] => {
  return dueDates.map((dueDate: string) => ({
    identifier: "ANNUAL_FILING",
    dueDate,
  }));
```

and it gets called in stuff like the `getAnnualFilings` test like:
```
taxFilingData: {
    ...postedUserData.taxFilingData,
    filings: generateAnnualFilings(["2021-03-01","2022-03-01","2023-03-01"]),
  },
```

Reply:
I think that makes sense. I'm not super torn one way or another. Included.